src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -728,28 +739,41 @@ WorkerShardStats(char *nodeName, uint32 nodePort, Oid relationId, char *shardNam
 	appendStringInfo(partitionValueQuery, SHARD_RANGE_QUERY,
 					 partitionColumnName, partitionColumnName, shardName);
 
-	queryOK = MultiClientExecute(connectionId, partitionValueQuery->data,
-								 &queryResult, &rowCount, &columnCount);
-	if (!queryOK)
+	querySent = SendRemoteCommand(connection, partitionValueQuery->data);
+	if (!querySent)
+	{
+		ReportConnectionError(connection, WARNING);
+		return false;
+	}
+
+	queryResult = GetRemoteCommandResult(connection, doRaiseInterrupts);
+	if (!IsResponseOK(queryResult))
 	{
-		MultiClientDisconnect(connectionId);
+		ReportResultError(connection, queryResult, WARNING);
+		PQclear(queryResult);
+		ForgetResults(connection);
 		return false;
 	}
```

Comment:
Might make more sense to add a wrapper for PQexec() that uses Send/GetRemoteCommandResult